sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala

```diff
@@ -317,12 +317,11 @@ private[orc] case class OrcTableScan(
       classOf[OrcInputFormat]
         .asInstanceOf[Class[_ <: MapRedInputFormat[NullWritable, Writable]]]
 
-    val rdd = sqlContext.sparkContext.hadoopRDD(
+    val rdd = new HadoopRDD(sqlContext.sparkContext,
```

Comment:
can you add a comment to say we're creating a HadoopRDD here directly to bypass closure cleaning